File: gateway-server/src/main/java/org/lucycato/gatewayserver/application/service/AuthGlobalFilterService.java
package org.lucycato.gatewayserver.application.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import org.lucycato.gatewayserver.application.port.in.AuthGlobalFilterUseCase;
import org.lucycato.gatewayserver.application.port.out.UserAuthPort;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
@RequiredArgsConstructor
public class AuthGlobalFilterService implements AuthGlobalFilterUseCase {

    private final ObjectMapper objectMapper;
    private final UserAuthPort userAuthPort;

    @Override
    public Mono<String> parseJwtToAdminUserJson(String token) {
        if (token != null && token.startsWith("Bearer ")) {
            return userAuthPort.getAdminUserAuthByToken(token)
                    .flatMap(adminUserAuth -> Mono.fromCallable(() -> objectMapper.writeValueAsString(adminUserAuth)));
        } else {
            return Mono.empty();
        }
    }

    @Override
    public Mono<String> parseJwtToAppUserJson(String token) {
        if (token != null && token.startsWith("Bearer ")) {
            return userAuthPort.getAppUserAuthByToken(token)
                    .flatMap(appUserAuth -> Mono.fromCallable(() -> objectMapper.writeValueAsString(appUserAuth)));
        } else {
            return Mono.empty();
        }
    }
}

File: gateway-server/src/main/kotlin/org/lucycato/gatewayservice/application/service/AuthGlobalFilterService.kt
package org.lucycato.gatewayservice.application.service

import com.fasterxml.jackson.databind.ObjectMapper
import org.lucycato.gatewayservice.application.port.`in`.AuthGlobalFilterUseCase
import org.lucycato.gatewayservice.application.port.out.UserAuthPort
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class AuthGlobalFilterService(
        private val objectMapper: ObjectMapper,
        private val userAuthPort: UserAuthPort
) : AuthGlobalFilterUseCase {
    override fun parseJwtToAdminUserJson(token: String?): Mono<String> = if (token != null && token.startsWith("Bearer ")) {
        userAuthPort.getAdminUserAuthByToken(token)
                .flatMap { adminUserAuth ->
                    Mono.fromCallable { objectMapper.writeValueAsString(adminUserAuth) }
                }
    } else {
        Mono.empty()
    }

    override fun parseJwtToAppUserJson(token: String?): Mono<String> = if (token != null && token.startsWith("Bearer ")) {
        userAuthPort.getAppUserAuthByToken(token)
                .flatMap { appUserAuth ->
                    Mono.fromCallable { objectMapper.writeValueAsString(appUserAuth) }
                }
    } else {
        Mono.empty()
    }

}